src/dsp_tools/commands/xmlupload/resource_create_client.py

```diff
@@ -151,30 +155,25 @@ def make_values(p: XMLProperty) -> list[dict[str, Any]]:
         last_prop_name = None
 
         str_value_to_serialiser_mapper = {
-            "color": SerialiseColor,
-            "geoname": SerialiseGeoname,
-            "time": SerialiseTime,
-            "uri": SerialiseURI,
+            "color": (SerialiseColor, StringTransformer()),
+            "date": (SerialiseDate, DateTransformer()),
+            "decimal": (SerialiseDecimal, DecimalTransformer()),
+            "geometry": (SerialiseGeometry, GeometryTransformer()),
+            "geoname": (SerialiseGeoname, StringTransformer()),
+            "time": (SerialiseTime, StringTransformer()),
+            "uri": (SerialiseURI, StringTransformer()),
```

Comment:
There are 2 things which I think are not optimal:

1. Every time that make_values() is called, this dictionary is built again. It would make more sense if it lives on the top level, or even better in the `value_transformers.py`. (See my PR https://github.com/dasch-swiss/dsp-tools/pull/1281)
2. The tuple `(SerialiseURI, StringTransformer())` is untyped, so you have to do a "blind" unpacking operation like `serialiser, transformer = str_value_to_serialiser_mapper[val_type]`. Alternatively, you could use a DTO (data transfer object), which encapsulates this info. For example like this:

```
@dataclass
class TransformationSteps:
    serialiser: ValueSerialiser
    transformer: ValueTransformer
```

Then, you could annotate this dict with `dict[str, TransformationSteps]`. But I admit this this perhaps an overkill, you can also leave it as it is.